core/commands/storage.go

```diff
@@ -147,34 +144,63 @@ Receive proofs as collateral evidence after selected nodes agree to store the fi
 
 		// get other node's public key as address
 		// create channel between them
-		var pid peer.ID
-		for _, str := range peers {
-			_, pid, err = ParsePeerParam(str)
-			if err != nil {
-				return fmt.Errorf("failed to parse peer address '%s': %s", str, err)
-			}
-			peerPubKey, err := pid.ExtractPublicKey()
-			if err != nil {
-				return fmt.Errorf("fail to extract public key from peer ID: %s", err)
-			}
-			channelID, err = initChannel(req.Context, selfPubKey, selfPrivKey, peerPubKey, price)
-			if err != nil {
-				continue
+		var wg sync.WaitGroup
+		wg.Add(len(fileHash))
+		chunkChan := make(chan *storage.Chunks)
+		go func() {
+			wg.Wait()
+			close(chunkChan)
+		}()
+		// FIXME: Fake multiple chunk hash as chunk hash
+		for i, chunk := range req.Arguments {
+			go func(chunkHash string, i int) {
+				defer wg.Done()
+				_, hostPid, err := ParsePeerParam(peers[i])
+				if err != nil {
+					chunkChan <- &storage.Chunks{
+						ChunkHash: chunkHash,
+						Err:       err,
+					}
+					return
+				}
+				peerPubKey, err := hostPid.ExtractPublicKey()
+				if err != nil {
+					chunkChan <- &storage.Chunks{
+						ChunkHash: chunkHash,
+						Err:       err,
+					}
+					return
+				}
+				channelID, err = initChannel(req.Context, selfPubKey, selfPrivKey, peerPubKey, price)
+				if err != nil {
+					chunkChan <- &storage.Chunks{
+						ChunkHash: chunkHash,
+						Err:       err,
+					}
+					return
+				}
+				_, err = p2pCall(n, hostPid, "/storage/upload/init", ssID, strconv.FormatInt(channelID.Id, 10), chunkHash)
+				if err != nil {
+					chunkChan <- &storage.Chunks{
+						ChunkHash: chunkHash,
+						Err:       err,
+					}
+					return
+				} else {
+					chunkChan <- &storage.Chunks{
+						ChunkHash: chunkHash,
+						Err:       nil,
+					}
+				}
+			}(chunk, i)
+		}
+		for range req.Arguments {
+			chunk := <- chunkChan
+			ss.Chunk = append(ss.Chunk, *chunk)
```

Comment:
Can you make `Chunk` type `[]*Chunks` so it's more efficient than copying `*chunk`?